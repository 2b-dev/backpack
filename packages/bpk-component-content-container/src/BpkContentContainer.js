import React, { PropTypes } from 'react'

import './bpk-content-container.scss'

const BpkContentContainer = ({ children, tagName }) => {
  const TagName = tagName
  return <TagName className='bpk-content-container' children={children} />
}

BpkContentContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  tagName: PropTypes.oneOf([ 'article', 'aside', 'div', 'main', 'section' ])
}

BpkContentContainer.defaultProps = {
  tagName: 'div'
}

export default BpkContentContainer

