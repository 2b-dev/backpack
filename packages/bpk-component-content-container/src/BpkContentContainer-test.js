import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';
import BpkContentContainer from './BpkContentContainer';

describe('BpkContentContainer', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkContentContainer>
        <h1>Heading</h1>
        <p>My paragraph.</p>
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
          <li>Item 3
          </li>
        </ul>
      </BpkContentContainer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with a "bareHtml" attribute ', () => {
    const tree = renderer.create(
      <BpkContentContainer bareHtml>
        <h1>Heading</h1>
        <p>My paragraph.</p>
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
          <li>Item 3
          </li>
        </ul>
      </BpkContentContainer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
