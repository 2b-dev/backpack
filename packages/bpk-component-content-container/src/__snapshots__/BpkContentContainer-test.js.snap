exports[`BpkContentContainer should render correctly 1`] = `
<div
  className="bpk-content-container">
  <h1>
    Heading
  </h1>
  <p>
    My paragraph.
  </p>
  <ul>
    <li>
      Item 1
    </li>
    <li>
      Item 2
    </li>
    <li>
      Item 3
    </li>
  </ul>
</div>
`;
